File: miscellaneous/hci.js
// https://www.acmicpc.net/problem/16139

const stdin = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : __dirname + "/../input.txt")
  .toString()
  .trim()
  .split("\n");
const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const s = input();
const q = input();
const answer = [];
const cumulative = [];

for (let i in s) {
  let temp;
  if (!cumulative.length) temp = Array(26).fill(0);
  else temp = [...cumulative.at(-1)];

  temp[s[i].charCodeAt(0) - 97] += 1;
  cumulative.push(temp);
}

for (let n = 0; n < q; n++) {
  let [alpha, from, to] = input().split(" ");
  answer.push(
    cumulative[to][alpha.charCodeAt(0) - 97] - (Number(from) && cumulative[from - 1][alpha.charCodeAt(0) - 97])
  );
}

console.log(answer.join("\n"));

File: miscellaneous/hci.py
# https://www.acmicpc.net/problem/16139

import sys
input = sys.stdin.readline

s = input().strip()
answer = []
acc = [[0 for i in range(len(s))] for i in range(26)]
set = set()

for _ in range(int(input())):
    alpha, start, end = input().split();
    start, end = int(start), int(end)
    if alpha in set:
        answer.append(acc[ord(alpha) - 97][end] - (acc[ord(alpha) - 97][start - 1] if start else 0))
    else:
        set.add(alpha)
        array = acc[ord(alpha)-97]
        for i, c in enumerate(s):
            if i: array[i] = array[i-1] + (c == alpha)
            else: array[0] = 0 + (c == alpha)
        answer.append(array[end] - (array[start-1] if start else 0))

print(*answer, sep="\n")